internal/parseutil/filereader_test.go

```diff
@@ -0,0 +1,48 @@
+package parseutil
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestLineResolution(t *testing.T) {
+	fr := &FileReader{}
+
+	file1 := "file1"
+	file2 := "file2"
+
+	addFile := func(fileName string, code string) {
+		if nil != fr.AddFile(fileName, strings.NewReader(code)) {
+			panic("should not happen")
+		}
+	}
+
+	addFile(file1, `BEGIN {
+print f(1)
+}`)
+	addFile(file2, `function f(x) {
+print x
+}`)
+	if len(fr.files) != 2 {
+		t.Errorf("must be 2 files")
+	}
+
+	{
```

Comment:
Maybe we can make these table-driven subtests with `t.Run`?

I think we want a few more tests, specifically no files, a zero length file, and files with and without a trailing `\n` on the last line.

Perhaps a table-driven test struct something like this:

```go
type test struct {
    // input:
    files []struct{ name, source string }
    line int
    // expected:
    path string
    fileLine int
}
```

And make sure we test `Source()` for completeness.